equinox/nn/conv.py

```diff
@@ -0,0 +1,227 @@
+import collections
+import math
+from itertools import repeat
+from typing import Optional, Sequence, Union
+
+import jax.numpy as jnp
+import jax.random as jrandom
+from jax.lax import conv_general_dilated
+
+from ..custom_types import Array
+from ..module import Module, static_field
+
+
+def _ntuple(n):
+    def parse(x):
+        if isinstance(x, collections.abc.Iterable):
+            return tuple(x)
+        return tuple(repeat(x, n))
+
+    return parse
+
+
+class Conv(Module):
+    num_spatial_dims: int = static_field()
+    weight: Array
+    bias: Optional[Array]
+    in_channels: int = static_field()
+    out_channels: int = static_field()
+    kernel_size: Union[int, Sequence[int]] = static_field()
+    stride: Union[int, Sequence[int]] = static_field()
+    padding: Union[int, Sequence[int]] = static_field()
+    dilation: Union[int, Sequence[int]] = static_field()
+    use_bias: bool = static_field()
+
+    def __init__(
+        self,
+        num_spatial_dims,
+        in_channels,
+        out_channels,
+        kernel_size,
+        stride=1,
+        padding=0,
+        dilation=1,
+        use_bias=True,
+        *,
+        key,
+        **kwargs,
+    ):
+        super().__init__()
```

Comment:
It would be best to pass `**kwargs` here for the sake of co-operative multiple inheritance.

Reply:
(Admittedly it looks like I've been inconsistent with this one myself in the rest of Equinox!)